src/main/java/com/iota/iri/storage/rocksDB/RocksDBPersistenceProvider.java

```diff
@@ -530,4 +548,56 @@ private void initClassTreeMap(List<ColumnFamilyDescriptor> columnFamilyDescripto
         classTreeMap = MapUtils.unmodifiableMap(classMap);
     }
 
+    private DBOptions createOptions(String logPath, String configFile) throws IOException {
+        DBOptions options = null;
+        File pathToLogDir = Paths.get(logPath).toFile();
+        if (!pathToLogDir.exists() || !pathToLogDir.isDirectory()) {
+            boolean success = pathToLogDir.mkdir();
+            if (!success) {
+                log.warn("Unable to make directory: {}", pathToLogDir);
+            }
+        }
+
+        int numThreads = Math.max(1, Runtime.getRuntime().availableProcessors() / 2);
+        RocksEnv.getDefault()
+            .setBackgroundThreads(numThreads, Priority.HIGH)
+            .setBackgroundThreads(numThreads, Priority.LOW);
+
+        if (configFile != null) {
+            File config = Paths.get(configFile).toFile();
+            if (config.exists() && config.isFile() && config.canRead()) {
+                Properties configProperties = new Properties();
+                InputStream stream = new FileInputStream(config);
+                try {
+                    configProperties.load(stream);
+                    options = DBOptions.getDBOptionsFromProps(configProperties);
+                } catch (IllegalArgumentException e) {
+                    log.warn("RocksDB configuration file is empty, falling back to default values");
+                } finally {
+                    stream.close();
+                }
```

Comment:
I always like to use "try with resources" e.g.
`try (InputStream stream = new FileInputStream(config))`

Just a bit less verbose. What you did works as well, but I'd rather we use the shorter way throughout the project.